src/terminal_chat_console_silent.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+Minetest
+Copyright (C) 2015 est31 <MTest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include <inttypes.h>
+#include "config.h"
+#if USE_CURSES
+#include "version.h"
+#include "terminal_chat_console_silent.h"
+#include "porting.h"
+#include "settings.h"
+#include "util/numeric.h"
+#include "util/string.h"
+#include "chat_interface.h"
+#include <fcntl.h>
+
+TerminalChatConsoleSilent g_term_console_silent;
+
+void TerminalChatConsoleSilent::initOfCurses()
+{
+	// Make cin non-blocking.
+	int flags = fcntl(0, F_GETFL, 0);
+	fcntl(0, F_SETFL, flags | O_NONBLOCK);
+}
+
+void TerminalChatConsoleSilent::deInitOfCurses()
+{
+	// no-op
+}
+
+void *TerminalChatConsoleSilent::run()
+{
+	BEGIN_DEBUG_EXCEPTION_HANDLER
+
+	// Inform the server of our nick
+	m_chat_interface->command_queue.push_back(
+		new ChatEventNick(CET_NICK_ADD, m_nick));
+
+	{
+		// Ensures that curses is deinitialized even on an exception being thrown
+		CursesInitHelper helper(this);
+
+		while (!stopRequested()) {
+
+			if (stopRequested())
+				break;
+
+			step();
+
+			// 10 / 1000 means you can only run 100 commands per second, the horror.
+			// If you are running 100 commands per second, you might want to 
+			// redesign the program interfacing with the server.
+			// This keeps the thread from maxing out the cpu core.
+			std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
no need for nonblock (which afaik could also cause problems with logging) and the sleeping.
you can simply `poll` with a timeout of e.g. 1s (so it still exits correctly)